### ISSUE TITLE
Ensure all groups/user creates in IAM Identity Store are via SCIM api

### ISSUE THREAD

**Issue by ChrisPates** [awslabs/ssosync#166]
**Is your feature request related to a problem? Please describe.**
To enable, other improvements the creation/update/delete of users and groups needs to be consistently carried out via the SCIM api endpoints and not mixed with the Identity Store API. This will allow sync entities to be differentiated from manually created users. The only partial exception would be where a manually created entity matches an entity to be synced, in which case it would be updated via the SCIM apis and switch from being a manually created entity to a synced one.

**Dependancies**
#141 
#142

**Comment by philomory** [comment]
Would having this implemented also avoid situations where the change of a group's "Display Name" in Google Workspace causes `ssosync` to delete its existing representation of that group and create a new group with the new name? I'm hoping the availability of `externalId` would prevent that?

**Comment by ChrisPates** [comment]
Yes, it would. Sadly the current code does not make use of the external id field and relies on the display name for groups and email address for users.

**Comment by ChrisPates** [comment]
Implemented in v2.5.0
